Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/reissuer/DccReissuer.kt

```diff
@@ -20,14 +24,39 @@ class DccReissuer @Inject constructor(
         InvalidHealthCertificateException::class
     )
     suspend fun startReissuance(dccReissuanceDescriptor: CertificateReissuance) {
-        val response = dccReissuanceProcessor.requestDccReissuance(dccReissuanceDescriptor)
-        val dccReissuance = response.dccReissuances.find { issuance ->
-            issuance.relations.any { r -> r.action == "replace" && r.index == 0 }
-        } ?: throw DccReissuanceException(ErrorCode.DCC_RI_NO_RELATION)
-
-        dccSwapper.swap(
-            dccReissuance = dccReissuance,
-            certificateToRecycle = dccReissuanceDescriptor.certificateToReissue
+        dccReissuanceDescriptor.certificates?.forEach {
+            reissue(it)
+        }
+    }
+
+    suspend fun reissue(item: CertificateReissuanceItem) {
+        val allQrCodes = mutableListOf(item.certificateToReissue) + item.accompanyingCertificates
+        val response = dccReissuanceServer.requestDccReissuance(
+            action = item.action,
+            certificates = allQrCodes.map { it.certificateRef.barcodeData }
         )
+
+        response.dccReissuances.forEach { issuance ->
+            register(issuance.certificate)
+            issuance.relations.filter { relation ->
+                relation.action == "replace"
+            }.forEach {
+                moveToBin(
+                    allQrCodes[it.index].certificateRef.barcodeData,
+                )
```

Comment:
Ok, I will look into how I can solve that. I changed it because the API and tech spec allows several replacements, there is no restriction to 1:1 replacement

Reply:
There is a list now anyways, so I have to tackle that either way

Reply:
will be addressed with https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13009